missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java

```diff
@@ -680,6 +713,37 @@ public void setArena(Arena arena) {
         ready = true;
     }
 
+    private void createInnerGameArea() {
+
+        // If both spawn points are on the X coordinate, the "length" of the Inner Game-Area 
+        // is the distance on X and the "width" of the Inner Game-Area is the width of the 
+        // (major) GameArea. Otherwise, it is the other way around. It is therefore 
+        // independent of the rotation of the GameArea.
+        
+        Location position1, position2;
+        int spawnPos1, spawnPos2;
+        
+        if (team1.getSpawn().getBlockX() == team2.getSpawn().getBlockX()) {
```

Comment:
This will not work on maps, which have spawn points which are not directly opposite (I have played on some). 
Therefore we need a more clever way to determine the inner game area:

Option A (old implementation):
We use my old method, constructing two [planes](https://upload.wikimedia.org/wikipedia/commons/thumb/f/f9/Plane_equation_qtl3.svg/1200px-Plane_equation_qtl3.svg.png) containing each spawn points. Then we use the the vector from spawn team 1 to spawn team 2 and use it determine the direction of the plane (this vector is perpendicular to the planes). 
When a missile gets placed, we check if the point is in between the two planes and in the game area (thus having a proper bound to the left/right).

Option B:
We construct a flat area with the two spawn points as our corners. Then we check, which side is longer (X or Y axis) and expand the shorter axis to the bounds of the game area.

Reply:
Very nice fix. Also with the use of the Direction enum, I like it!